xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/DecimalField.java

```diff
@@ -46,4 +50,46 @@ public void setValue(Object value) throws InvalidOutcomeException {
 
         super.setData(value.toString());
     }
+
+    @Override
+    public void setNgDynamicFormsValidators(JSONObject validators) {
+        //locale specific separators could be used, but it should be based on the locale of the browser
+        //char separator = new DecimalFormatSymbols(Locale.getDefault(Locale.Category.FORMAT)).getDecimalSeparator();
+        char separator = Gateway.getProperties().getString("Webui.decimal.separator", ".").charAt(0);
+
+        super.setNgDynamicFormsValidators(validators);
+
+        if (StringUtils.isNotBlank(pattern)) {
+            validators.put("pattern", pattern);
+        }
+        else {
+            Integer totalDigits = null;
+            Integer fractionDigits = null;
+
+            if(contentType.hasFacet(Facet.TOTALDIGITS)) {
+                Facet f = contentType.getFacet(Facet.TOTALDIGITS);
+                totalDigits = Integer.valueOf(f.getValue());
+            }
+
+            if(contentType.hasFacet(Facet.FRACTIONDIGITS)) {
+                Facet f = contentType.getFacet(Facet.FRACTIONDIGITS);
+                fractionDigits = Integer.valueOf(f.getValue());
+            }
+
+            if (totalDigits == null && fractionDigits == null) {
+                //default validator for any decimal field
+                validators.put("pattern", "^\\d+\\" + separator + "?\\d+$");
+            }
+            else if (totalDigits != null) {
+                if (fractionDigits == null) {
+                    validators.put("pattern", "^\\d{0," + totalDigits + "}$");
+                }
+                else
+                    validators.put("pattern", "^\\d{0," + (totalDigits - fractionDigits) + "}\\" + separator + "?\\d{0," + fractionDigits + "}$");

```

Comment:
Use the pattern I suggested to enforce the exact number of decimals as specified in the user requirements.

Reply:
The suggested pattern does not work, when `totalDigits=5` and `fractionDigits=2` , it shall accept 123.45 and 1234.5

Reply:
No, it shall not accept 1234.5 based on Enrico's user requirements. It's a fixed number of decimals that he wants enforced. https://jsfiddle.net/wphb21vf/1/

Reply:
You did not get my point. The pattern has to do the same restriction implemented for XSD validation, otherwise there will be errors that is difficult to explain to the user.

Reply:
OK, so why the hell are we implementing this? This does not implement the requirements.